src/Kafka/ManageKafkaPermissions/components/SelectAccount.tsx

```diff
@@ -0,0 +1,194 @@
+
+export type Validated<T> = {
+  value: T;
+  validated?: ValidatedOptions;
+  errorMessage?: string;
+};
+
+export type SelectAccountProps = {
+  id: string | undefined | SelectOptionObject;
+  onChangeAccount: React.Dispatch<
+    React.SetStateAction<string | undefined | SelectOptionObject>
+  >;
+  accounts: Account[];
+  onEscapeModal: (closes: boolean) => void;
+  validated: ValidatedOptions;
+  onChangeValidation: (value: ValidatedOptions) => void;
+};
+
+export const SelectAccount: React.VFC<SelectAccountProps> = ({
+  onChangeAccount,
+  id,
+  accounts,
+  onEscapeModal,
+  validated,
+  onChangeValidation,
+}) => {
+  const { t } = useTranslation(["manage-kafka-permissions"]);
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+
+  const onToggle = (newState: boolean) => {
+    if (newState) {
+      onEscapeModal(false);
+    } else onEscapeModal(true);
+
+    setIsOpen(newState);
+  };
+
+  const clearSelection = () => {
+    onChangeAccount(undefined);
+    onChangeValidation(ValidatedOptions.error);
+    setIsOpen(false);
+  };
+  const serviceAccountOptions = () => {
+    const serviceAccountsLength = accounts.filter(
+      (account) => account.principalType === PrincipalType.ServiceAccount
+    ).length;
+    return serviceAccountsLength
+      ? accounts
+          .filter(
+            (principal) =>
+              principal.principalType === PrincipalType.ServiceAccount
+          )
+          .sort((a, b) =>
+            a.displayName && b.displayName
+              ? a.displayName.localeCompare(b.displayName)
+              : -1
+          )
+          .map((principal, index) => (
+            <SelectOption
+              key={index}
+              value={principal.id}
+              description={principal.displayName}
+            >
+              {principal.id}
+            </SelectOption>
+          ))
+      : [
+          <SelectOption isNoResultsOption={true} isDisabled={true} key={1}>
+            {t("manage_permissions_dialog.no_results_found")}
+          </SelectOption>,
+        ];
+  };
+
+  const userAccountOperations = () => {
+    const userAccountsLength = accounts.filter(
+      (account) => account.principalType === PrincipalType.UserAccount
+    ).length;
+    return userAccountsLength
+      ? accounts
+          .filter(
+            (principal) => principal.principalType === PrincipalType.UserAccount
+          )
+          .map((principal, index) => (
+            <SelectOption
+              key={index}
+              value={principal.id}
+              description={principal.displayName}
+            >
+              {principal.id}
+            </SelectOption>
+          ))
+      : [
+          <SelectOption isNoResultsOption={true} isDisabled={true} key={1}>
+            {t("manage_permissions_dialog.no_results_found")}
+          </SelectOption>,
+        ];
+  };
+
+  const onSelect: SelectProps["onSelect"] = (_, selection) => {
+    onChangeAccount(selection);
```

Comment:
@riccardo-forina  The reason I added SelectOptionObject type instead of just going with a string|undefined type for the state of value is the type error I'm getting here. 
**_Argument of type 'string | SelectOptionObject' is not assignable to parameter of type 'string | undefined'.
  Type 'SelectOptionObject' is not assignable to type 'string'_**
Any suggestions on how I can remove that?

Reply:
In this case it's ok to do a cast `as string` since you are feeding strings to the select.